File: basics/gcd.py

# Greatest Common Divisor (GCD)
# Largest positive intger that divides a number 

def gcd(a,b) :
    while (b!=0):
        a, b = b, a % b
    return a

if __name__ == "__main__" :
    a = int(input("Enter a number:"))
    b = int(input("Enter another number:"))
    res = gcd(a,b)
    print(f"\nThe gcd of {a} and {b} is:", res)

File: basics/gcd.cpp

#include <bits/stdc++.h>

using namespace std;

int gcd(int a, int b) {
    int temp;
    while (b != 0) {
        temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main() {
    int a = 0;
    int b = 15;
    int res = gcd(a,b);
    cout << "The gcd of " << a << " and " << b << " is " << res;
    return 0;
}